pallets/stellar-relay/src/lib.rs

```diff
@@ -475,19 +474,26 @@ pub mod pallet {
 					current_organizations,
 				)
 				.map_err(|_| Error::<T>::BoundedVecCreationFailed)?;
-			OldOrganizations::<T>::put(current_organizations);
 
 			NewValidatorsEnactmentBlockHeight::<T>::put(enactment_block_height);
 
 			let new_validator_vec =
 				BoundedVec::<ValidatorOf<T>, T::ValidatorLimit>::try_from(validators)
 					.map_err(|_| Error::<T>::BoundedVecCreationFailed)?;
-			Validators::<T>::put(new_validator_vec);
 
 			let new_organization_vec =
 				BoundedVec::<OrganizationOf<T>, T::OrganizationLimit>::try_from(organizations)
 					.map_err(|_| Error::<T>::BoundedVecCreationFailed)?;
-			Organizations::<T>::put(new_organization_vec);
+
+			//update only when new organization or validators not equal to old organization or validators
+			if new_organization_vec != current_organizations ||
+				new_validator_vec != current_validators
+			{
+				OldValidators::<T>::put(current_validators.clone());
+				OldOrganizations::<T>::put(current_organizations.clone());
+				Validators::<T>::put(new_validator_vec.clone());
+				Organizations::<T>::put(new_organization_vec.clone());
+			}
```

Comment:
I think we should do this differently. Since we need to add other checks I'd propose we do something like this:
```
// Check if validators and organizations change
ensure!(new_organization_vec != current_organizations && new_validator_vec != current_validators, Error::<T>::SameValidatorsAndOrganizations);

// Check for duplicates in the validators vec
// TODO figure out the best way to check for duplicates in this vec (should not be too computationally expensive)
// return an appropriate error if we encounter a duplicate

// Check for duplicates in the organizations vec
// TODO figure out the best way to check for duplicates in this vec (should not be too computationally expensive)
// return an appropriate error if we encounter a duplicate

// Set the values after all checks passed
OldValidators::<T>::put(current_validators.clone());
OldOrganizations::<T>::put(current_organizations.clone());
Validators::<T>::put(new_validator_vec.clone());
Organizations::<T>::put(new_organization_vec.clone());
```

Note that I deliberately used an `&&` operator instead of `||` because it might be that we want to update just an organization **or** something about the validators. If we used `||` then we'd always have to change something

Reply:
@ebma 
add validation for duplication organisation id and public_key in validators.
fix unit tests to pass different organisation id and public key.